Codecademy/client-modules
# chore(dependencies): Typescript 4.1.3 [WEB-1253]

### Overview

Typescript 4.1.x introduces some extremely useful features for us that should improve our ability to make the generated types from `gamut-system` more performant.
- Codecademy: https://github.com/codecademy-engineering/Codecademy/pull/21549

This enables support for Typescript 4.1.3.


**Dependencies**
- `prettier`: 2.1.2 => 2.2.1 
- `react-use`: 15.3.4 => 15.3.8 
- `typescript-eslint`: 4.4.1 => 4.14.1

**Storybook**
- Fixes storybook to use our own babel config and docgen options.
- Fixed tsconfig to pick up the correct files.


### PR Checklist

- [x] Related to JIRA ticket: WEB-1253

### Testing
1. Deploy locally
2. Run `yarn`
3. Open VSCode (after restarting it) and a component file.

Add: 
```tsx
type Props = {
  [Key in 'Click' | 'Change' as `on${ Key }` ]:  () => void
}
```
4. Save the file and see that prettier formats it correctly
5. Add the type to an existing component.
6. Run `yarn start` and see that storybook builds.
7. Check the components story page to see that it has the correct types in the props table.
